runtime_scan/pkg/orchestrator/common/queue.go

```diff
@@ -100,8 +107,6 @@ func (q *Queue[T]) Dequeue(ctx context.Context) (T, error) {
 		default:
 		}
 	}
-
-	q.l.Lock()
 	defer q.l.Unlock()
```

Comment:
This can be moved on line 98

Reply:
This defer applies to two lock lines, if len(q.queue) == 0 then it defers the unlock of the Lock @97, if len(q.queue) > 0 then it defers the Lock @86 so I can't move this up otherwise we won't unlock the case when there are already items in the queue.